### DISCUSSION TITLE
Support ESP32-ADF from Olimex

### DISCUSSION THREAD

**Discussion by zeridon** [sle118/squeezelite-esp32#240] [Ideas]
I would like to have ESP32-ADF from Olimex supported as a targed (sdkconfig)

The board itself is ESP AudioKit very similar to A1S from AI Thinkerer.
Specifications:
 * ESP32-WROVER-B with 8MB PSRAM, 4MB Flash
 * ES8388 + PAM8003
 * SD Card, UEXT connector (custom), LiPo charger, microphones, ir, etc

Product Page: https://www.olimex.com/Products/IoT/ESP32/ESP32-ADF/open-source-hardware
Schematic: https://github.com/OLIMEX/ESP32-ADF/tree/master/HARDWARE/ESP32-ADF%20Rev.C (kicad + schematic in PDF)

I have a board available for testing and am not afraid to do light meddling with code.

Alternatively point me where to start playing with config and will try to do it myself

**Comment by AxelJust** [comment]
I have a Olimex running.
It is the identical configuration as with a Lyrat

**Comment by zeridon** [comment]
@AxelJust Do you mind sharing your configs

**Comment by c-OOL-io** [comment]
Here you can find 2 fully working configs for the OLiMEX ESP32-ADF board.
[https://squeezeesp32.blogspot.com/p/supported-hardware.html](https://squeezeesp32.blogspot.com/p/supported-hardware.html  )

There you will find also some more fully working NVS hardware configs of the following player, boards/dev kits.
Here the current full list. They could be helpful to add even these configs to one of the next FW2.x releases. ;)

    Makerfabs_MAKT32+MPAUDIO_DAC+OLED-only
    Makerfabs_MAKT32CL+MPAUDIO_DAC+LCD-only
    OLiMEX_ESP32ADF_DAC-only
    OLiMEX_ESP32ADF_DAC+OLED-I2C-SH1106-128x64
    raspiaudio_ESP_Muse_Mn_Cast_hardware-only

**Comment by sle118** [comment]
I'll try to include these configurations to the 4.3 version drop down list of known hardware

**Comment by c-OOL-io** [comment]
The config of the MAKT32 (OLED) screen has changed. Therefore the config is now on V1.1.
Only change, i2c_config": ...,port=0  instead port=1. Because port 1 seams to have sometimes issues with positioning the elements on the display in the right way/order. Which seems to be solved when using port 0.

    Makerfabs_MAKT32+MPAUDIO_DAC+OLED-only_V1.1
    Makerfabs_MAKT32CL+MPAUDIO_DAC+LCD-only_V1.0
    OLiMEX_ESP32ADF_DAC-only_V1.0
    OLiMEX_ESP32ADF_DAC+OLED-I2C-SH1106-128x64_V1.0
    raspiaudio_ESP_Muse_Mn_Cast_hardware-only_V1.0

**Comment by sle118** [comment]
Since you have access to these boards, could you share your config for them?

There's the format that the presets uses
https://gist.github.com/sle118/dae585e157b733a639c12dc70f0910c5

I believe you can simply create the config section by exporting your configuration to a file from the NVS menu, and removing all NVS entries that aren't specific to a given board (e.g. only keeping gpio related entries, d'ac control set, etc)

**Comment by philippe44** [comment]
Can we close this one?

**Comment by sle118** [comment]
I think I'll move that one to discussions, since there seems to be useful info here
